marlowe-playground-client/src/MarloweEditor/State.purs

```diff
@@ -131,59 +117,69 @@ handleAction _ (InitMarloweProject contents) = do
 
 handleAction _ (SelectHole hole) = assign _selectedHole hole
 
-handleAction settings AnalyseContract =
-  void
-    $ runMaybeT do
-        contents <- MaybeT $ editorGetValue
-        contract <- hoistMaybe $ parseContract' contents
-        -- when editor and simulator were together the analyse contract could be made
-        -- at any step of the simulator. Now that they are separate, it can only be done
-        -- with initial state
-        let
-          emptySemanticState = emptyState zero
-        assign _analysisState (WarningAnalysis Loading)
-        response <- lift $ checkContractForWarnings contract emptySemanticState
-        assign _analysisState (WarningAnalysis response)
-  where
-  checkContractForWarnings contract state = runAjax $ (flip runReaderT) settings (Server.postMarloweanalysis (MSReq.Request { onlyAssertions: false, contract, state }))
-
-handleAction settings AnalyseReachabilityContract =
-  void
-    $ runMaybeT do
-        contents <- MaybeT $ editorGetValue
-        contract <- hoistMaybe $ parseContract' contents
-        -- when editor and simulator were together the analyse contract could be made
-        -- at any step of the simulator. Now that they are separate, it can only be done
-        -- with initial state
-        let
-          emptySemanticState = emptyState zero
-        newReachabilityAnalysisState <- lift $ startReachabilityAnalysis settings contract emptySemanticState
-        assign _analysisState (ReachabilityAnalysis newReachabilityAnalysisState)
-
-handleAction settings AnalyseContractForCloseRefund =
+handleAction settings AnalyseContract = runAnalysis $ analyseContract settings
+
+handleAction settings AnalyseReachabilityContract = runAnalysis $ analyseReachability settings
+
+handleAction settings AnalyseContractForCloseRefund = runAnalysis $ analyseClose settings
+
+handleAction _ Save = pure unit
+
+runAnalysis ::
+  forall m.
+  MonadAff m =>
+  (Contract -> HalogenM State Action ChildSlots Void m Unit) ->
+  HalogenM State Action ChildSlots Void m Unit
+runAnalysis doAnalyze =
   void
     $ runMaybeT do
         contents <- MaybeT $ editorGetValue
         contract <- hoistMaybe $ parseContract' contents
-        -- when editor and simulator were together the analyse contract could be made
-        -- at any step of the simulator. Now that they are separate, it can only be done
-        -- with initial state
-        let
-          emptySemanticState = emptyState zero
-        newCloseAnalysisState <- lift $ startCloseAnalysis settings contract emptySemanticState
-        assign _analysisState (CloseAnalysis newCloseAnalysisState)
-
-handleAction _ Save = pure unit
+        lift
+          $ do
+              doAnalyze contract
+              lintText contents
```

Comment:
why are you doing linting on analysis now?

Reply:
this solves a bug that we found with @palas... We show the "unrechable" analysis results in three places, in the Static analysis tab as a list (not very useful at the moment) and in the warning tab and as a linting marker. For the last two, we need to run the linting with the updated static analysis